app/services/orchestration_template_dialog_service.rb

```diff
@@ -213,7 +213,7 @@ def add_parameter_field(parameter, group, position)
   end
 
   def create_parameter_dropdown_list(parameter, group, position, dropdown)
-    dropdown_list = dropdown.allowed_values.collect { |v| [v, v] }
+    dropdown_list = dropdown.allowed_values.collect { |v| v.kind_of?(Hash) ? [v.keys.first, v[v.keys.first]] : [v, v] }
```

Comment:
We don't need to make it too complicated by allowing mixed types in the array. Ruby hash always keeps the order.